src/debugging/coreclr/LocalAspNetCoreSslManager.ts

```diff
@@ -0,0 +1,139 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import { MessageItem } from 'vscode';
+import { ext } from '../../extensionVariables';
+import { PlatformOS } from '../../utils/platform';
+import { quickPickProjectFileItem } from '../../utils/quick-pick-file';
+import { quickPickWorkspaceFolder } from '../../utils/quickPickWorkspaceFolder';
+import { ProcessProvider } from './ChildProcessProvider';
+import { DotNetClient, TrustState } from './CommandLineDotNetClient';
+import { OSProvider } from './LocalOSProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+
+export type SecretsFolders = {
+    certificateFolder: string;
+    userSecretsFolder: string;
+}
+
+export interface AspNetCoreSslManager {
+    trustCertificateIfNecessary(): Promise<void>;
+    exportCertificateIfNecessary(projectFile: string | undefined, certificateExportPath: string | undefined): Promise<void>;
+    getHostSecretsFolders(): SecretsFolders;
+    getContainerSecretsFolders(platform: PlatformOS): SecretsFolders;
+}
+
+export class LocalAspNetCoreSslManager implements AspNetCoreSslManager {

```

Comment:
Before you get comfortable with this pattern (Provider/Manager classes being passed around) - I want to quote [what I said](https://github.com/microsoft/vscode-docker/pull/500#pullrequestreview-163939080) to Phil when he first introduced it to this repo:

> I'm still not a huge fan of the Provider pattern for the following reasons:
> 1. I think it makes the code unnecessarily complicated
> 1. I think tests are more effective if they're running against the 'real thing'
> 1. I think you might be using patterns that were helpful testing against VS but are less necessary when testing VS Code (which is a lot easier from my experience)

It feels like the debugging code is off on it's own little world and not really integrated with the rest of the codebase. Regardless of what pattern we choose, I think the repo should be consistent. You obviously don't have to fix it for this PR, but I just want to make sure you're aware of this. If this was one-off work I might not care as much, but seems like y'all will be contributing more regularly now.

Reply:
I'd tend to agree. You end up with a very unwieldy factory in registerDockerDebug...ts, to construct a million and one objects, many of which don't actually have state and could be static methods.